pkg/operator/controller/dns/controller.go

```diff
@@ -559,6 +562,31 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			return nil, fmt.Errorf("failed to create GCP DNS provider: %v", err)
 		}
 		dnsProvider = provider
+	case configv1.IBMCloudPlatformType:
+		if infraStatus.ControlPlaneTopology != configv1.ExternalTopologyMode {
```

Comment:
I asked asked about a similar `if infraStatus.ControlPlaneTopology != configv1.ExternalTopologyMode` check in `createDNSProviderIfNeeded`, but I forgot to comment on this.  Can this check be done for all providers?

Reply:
Yes it can. Changed it accordingly.